eng/pipelines/templates/steps/cache-maven-repository.yml

```diff
@@ -1,15 +1,6 @@
 steps:
-- script: |
-    echo "##vso[task.setvariable variable=Maven.RepositoryPath;]%USERPROFILE%\.m2\repository"
-  condition: and(succeeded(), eq(variables['Agent.OS'], 'Windows_NT'))
-  displayName: Detecting Maven repository on Windows
-- script: |
-    echo "##vso[task.setvariable variable=Maven.RepositoryPath;]$HOME/.m2/repository"
-  condition: and(succeeded(), ne(variables['Agent.OS'], 'Windows_NT'))
-  displayName: Detecting Maven repository on Linux and macOS
-- task: CacheBeta@0
+- task: Cache@2
   inputs:
-    key: $(Agent.JobName)|$(CacheSalt)|$(Build.SourcesDirectory)/sdk/**/pom.xml|$(Build.SourcesDirectory)/pom*.xml
-    path: $(Maven.RepositoryPath)
-  displayName: 'Download/upload cache'
-  condition: and(succeeded(), ne(variables['EnableCaching'], 'false'))
\ No newline at end of file
+    key: 'maven | $(Agent.OS) | $(Build.SourcesDirectory)/eng/versioning/external_dependencies.txt'
```

Comment:
Does this key take into account the contents of the `external_dependencies.txt` file or just the name of it?

Reply:
It takes the content into account. It does a hash of the file and whenever this file changes the cache will be invalidated and recreated.